pwndbg/dbg/__init__.py

```diff
@@ -55,6 +86,30 @@ def bottom_frame(self) -> Frame:
         raise NotImplementedError()
 
 
+class MemoryMap:
+    """
+    A wrapper around a sequence of memory ranges
+    """
+
+    def is_qemu(self) -> bool:
+        """
+        Returns whether this memory map was generated from a QEMU target.
+        """
+        raise NotImplementedError()
+
+    def has_reliable_perms(self) -> bool:
+        """
+        Returns whether the permissions in this memory map are reliable.
+        """
+        raise NotImplementedError()
```

Comment:
Do we actually ever use `reliable_perms` or plan to use it anywhere?

Reply:
We do. There's a spot in the `vmmap` command that uses it for a warning. It's only really used there, but that was a way to convey that information to the `vmmap` command without having it rely on explicit GDB version checks after the port.